// @devfunlabs/web-sdk/dist/test/token.spec.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const mocha_1 = require("mocha/index.js");
const chai_1 = require("chai");
const pumpfun_1 = require("../pumpfun/index.cjs");
const dexscreener_1 = require("../dexscreener/index.cjs");
const token_1 = require("../token.cjs");
const TOKEN_MINT = 'AxGAbdFtdbj2oNXa4dKqFvwHzgFtW9mFHWmd7vQfpump';
const TOKEN_INFO = {
    contractAddress: TOKEN_MINT,
    symbol: 'HAT',
    name: 'TOP HAT',
    description: "Solana's first no-code AI agent launchpad.",
    creator: 'BXjEFPEDwgNrAdLRHb9kqLK4DrLtd3uQPcmiQpKPZTd5',
    twitter: 'https://x.com/TopHat_One',
    telegram: 'https://t.me/+DLUgJl8MCDBjNTFk',
    website: 'https://tophat.one',
    createdTime: 1730374103343,
    image: 'https://pump.mypinata.cloud/ipfs/QmPPhPbe9t5AKkBgBP4uon7aBdwLzukdS5PJoujC4YjJBj?img-width=256&img-dpr=2',
};
(0, mocha_1.describe)('Token info', () => {
    (0, mocha_1.it)('extract pumpfun token info', async () => {
        const info = await (0, pumpfun_1.tryFetchPumpfunToken)('AxGAbdFtdbj2oNXa4dKqFvwHzgFtW9mFHWmd7vQfpump');
        (0, chai_1.expect)(info?.contractAddress).to.equal(TOKEN_INFO.contractAddress);
        (0, chai_1.expect)(info?.symbol).to.equal(TOKEN_INFO.symbol);
        (0, chai_1.expect)(info?.name).to.equal(TOKEN_INFO.name);
        (0, chai_1.expect)(info?.description).to.equal(TOKEN_INFO.description);
        (0, chai_1.expect)(info?.creator).to.equal(TOKEN_INFO.creator);
        (0, chai_1.expect)(info?.twitter).to.equal(TOKEN_INFO.twitter);
        (0, chai_1.expect)(info?.telegram).to.equal(TOKEN_INFO.telegram);
        (0, chai_1.expect)(info?.website).to.equal(TOKEN_INFO.website);
        (0, chai_1.expect)(info?.createdTime).to.equal(TOKEN_INFO.createdTime);
        (0, chai_1.expect)(info?.image).to.equal(TOKEN_INFO.image);
        (0, chai_1.expect)(info?.marketcap).to.gt(0);
    });
    (0, mocha_1.it)('extract dexscreener token info', async () => {
        const info = await (0, dexscreener_1.tryFetchDexscreenerToken)('AxGAbdFtdbj2oNXa4dKqFvwHzgFtW9mFHWmd7vQfpump');
        (0, chai_1.expect)(info?.contractAddress).to.equal(TOKEN_INFO.contractAddress);
        (0, chai_1.expect)(info?.symbol).to.equal(TOKEN_INFO.symbol);
        (0, chai_1.expect)(info?.name).to.equal(TOKEN_INFO.name);
        (0, chai_1.expect)(info?.twitter).to.equal(TOKEN_INFO.twitter);
        (0, chai_1.expect)(info?.telegram).to.not.empty;
        (0, chai_1.expect)(info?.website).to.not.empty;
        (0, chai_1.expect)(info?.marketcap).to.gt(0);
        (0, chai_1.expect)(info?.price).to.gt(0);
        (0, chai_1.expect)(info?.buyTxIn24h).to.gt(0);
        (0, chai_1.expect)(info?.sellTxIn24h).to.gt(0);
        (0, chai_1.expect)(info?.volumeIn24h).to.gt(0);
        (0, chai_1.expect)(info?.priceChangeIn24h).to.not.null;
        (0, chai_1.expect)(info?.liquidityInUSD).to.gt(0);
    });
    (0, mocha_1.it)('extract token info by address', async () => {
        const info = await (0, token_1.fetchTokenInfoByAddress)('AxGAbdFtdbj2oNXa4dKqFvwHzgFtW9mFHWmd7vQfpump');
        console.info(info);
        (0, chai_1.expect)(info?.contractAddress).to.equal(TOKEN_INFO.contractAddress);
        (0, chai_1.expect)(info?.symbol).to.equal(TOKEN_INFO.symbol);
        (0, chai_1.expect)(info?.name).to.equal(TOKEN_INFO.name);
        (0, chai_1.expect)(info?.description).to.equal(TOKEN_INFO.description);
        (0, chai_1.expect)(info?.creator).to.equal(TOKEN_INFO.creator);
        (0, chai_1.expect)(info?.twitter).to.equal(TOKEN_INFO.twitter);
        (0, chai_1.expect)(info?.telegram).to.equal(TOKEN_INFO.telegram);
        (0, chai_1.expect)(info?.website).to.equal(TOKEN_INFO.website);
        (0, chai_1.expect)(info?.createdTime).to.equal(TOKEN_INFO.createdTime);
        (0, chai_1.expect)(info?.image).to.equal(TOKEN_INFO.image);
        (0, chai_1.expect)(info?.marketcap).to.gt(0);
        (0, chai_1.expect)(info?.price).to.gt(0);
    });
    (0, mocha_1.it)('extract token info by symbol', async () => {
        const info = await (0, token_1.fetchTokenInfoBySymbol)('HAT');
        console.info(info);
        (0, chai_1.expect)(info?.contractAddress).to.equal(TOKEN_INFO.contractAddress);
        (0, chai_1.expect)(info?.symbol).to.equal(TOKEN_INFO.symbol);
        (0, chai_1.expect)(info?.name).to.equal(TOKEN_INFO.name);
        (0, chai_1.expect)(info?.twitter).to.equal(TOKEN_INFO.twitter);
        (0, chai_1.expect)(info?.telegram).to.not.empty;
        (0, chai_1.expect)(info?.website).to.not.empty;
        (0, chai_1.expect)(info?.marketcap).to.gt(0);
        (0, chai_1.expect)(info?.price).to.gt(0);
        (0, chai_1.expect)(info?.buyTxIn24h).to.gt(0);
        (0, chai_1.expect)(info?.sellTxIn24h).to.gt(0);
        (0, chai_1.expect)(info?.volumeIn24h).to.gt(0);
        (0, chai_1.expect)(info?.priceChangeIn24h).to.not.null;
        (0, chai_1.expect)(info?.liquidityInUSD).to.gt(0);
    });
    (0, mocha_1.it)('fetch trending tokens', async () => {
        const tokens = await (0, pumpfun_1.fetchPumpFunTrends)();
        (0, chai_1.expect)(tokens.length).to.gt(10);
        (0, chai_1.expect)(tokens[0].name).to.not.empty;
    });
});
