// @devfunlabs/web-sdk/dist/devapp/wallet/utils/cssVariables.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createWalletCSSVariables = createWalletCSSVariables;
const DEFAULT_COLORS = {
    height: '48px',
    primaryColor: '#5ace90',
    radius: '2px',
    buttonBgColor: '#5ace90',
    buttonTextColor: '#1a1a28',
    buttonHoverBgColor: '#81D9AA',
    dropdownBgColor: '#28283C',
    dropdownTextColor: '#fff',
    dropdownHoverBgColor: '#383853',
};
function createWalletCSSVariables(variables = {}) {
    return {
        '--df-wallet-primary-color': variables.primaryColor || DEFAULT_COLORS.primaryColor,
        '--df-wallet-radius': variables.radius || DEFAULT_COLORS.radius,
        '--df-wallet-button-height': variables.height || DEFAULT_COLORS.height,
        '--df-wallet-button-bg-color': variables.buttonBgColor || DEFAULT_COLORS.buttonBgColor,
        '--df-wallet-button-text-color': variables.buttonTextColor || DEFAULT_COLORS.buttonTextColor,
        '--df-wallet-button-hover-bg-color': variables.buttonHoverBgColor || DEFAULT_COLORS.buttonHoverBgColor,
        '--df-wallet-dropdown-bg-color': variables.dropdownBgColor || DEFAULT_COLORS.dropdownBgColor,
        '--df-wallet-dropdown-text-color': variables.dropdownTextColor || DEFAULT_COLORS.dropdownTextColor,
        '--df-wallet-dropdown-hover-bg-color': variables.dropdownHoverBgColor || DEFAULT_COLORS.dropdownHoverBgColor,
    };
}
