// @devfunlabs/web-sdk/dist/devapp/wallet/utils/environment.mjs
export function getEnvironment() {
    if (typeof window === 'undefined' || typeof navigator === 'undefined') {
        return {
            isMobile: false,
            os: 'unknown',
            browser: 'unknown',
        };
    }
    const isMobile = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
    let os = 'unknown';
    const { userAgent } = navigator;
    if (/iPhone|iPad|iPod/i.test(userAgent)) {
        os = 'ios';
    }
    else if (/Android/i.test(userAgent)) {
        os = 'android';
    }
    else if (/Win/i.test(userAgent)) {
        os = 'windows';
    }
    else if (/Mac/i.test(userAgent)) {
        os = 'mac';
    }
    else if (/Linux/i.test(userAgent)) {
        os = 'linux';
    }
    let browser = 'unknown';
    if (/Chrome/i.test(userAgent) && !/Edg/i.test(userAgent)) {
        browser = 'chrome';
    }
    else if (/Firefox/i.test(userAgent)) {
        browser = 'firefox';
    }
    else if (/Safari/i.test(userAgent) && !/Chrome/i.test(userAgent)) {
        browser = 'safari';
    }
    else if (/Edg/i.test(userAgent)) {
        browser = 'edge';
    }
    else if (/Opera|OPR/i.test(userAgent)) {
        browser = 'opera';
    }
    return {
        isMobile,
        os,
        browser,
    };
}
export function getWalletInstallInfo(wallet) {
    const env = getEnvironment();
    if (env.isMobile) {
        if (env.os === 'ios' && wallet.app.ios) {
            return { url: wallet.app.ios, text: 'mobile app' };
        }
        if (env.os === 'android' && wallet.app.android) {
            return { url: wallet.app.android, text: 'mobile app' };
        }
    }
    else {
        const browserExtensions = {
            chrome: wallet.app.chrome,
            firefox: wallet.app.firefox,
            safari: wallet.app.safari,
            edge: wallet.app.edge,
            opera: wallet.app.opera,
        };
        if (env.browser !== 'unknown' && browserExtensions[env.browser]) {
            return { url: browserExtensions[env.browser], text: 'browser extension' };
        }
        for (const ext of Object.values(browserExtensions)) {
            if (ext) {
                return { url: ext, text: 'browser extension' };
            }
        }
        if (env.os === 'windows' && wallet.app.windows) {
            return { url: wallet.app.windows, text: 'desktop app' };
        }
        if (env.os === 'mac' && wallet.app.mac) {
            return { url: wallet.app.mac, text: 'desktop app' };
        }
        if (env.os === 'linux' && wallet.app.linux) {
            return { url: wallet.app.linux, text: 'desktop app' };
        }
    }
    const anyUrl = wallet.app.chrome
        || wallet.app.windows
        || wallet.app.mac
        || wallet.app.ios
        || wallet.app.android;
    if (anyUrl) {
        return { url: anyUrl, text: 'app' };
    }
    return { url: '#', text: 'app' };
}
