// @devfunlabs/web-sdk/dist/devbase/client/processor/base.cjs
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseProcessor = exports.DevbaseHeader = void 0;
const utils_1 = require("../../utils.cjs");
var DevbaseHeader;
(function (DevbaseHeader) {
    DevbaseHeader["SECRET"] = "X-Devbase-Secret";
    DevbaseHeader["USER_ID"] = "X-Devbase-UserId";
    DevbaseHeader["USER_TOKEN"] = "X-Devbase-UserToken";
})(DevbaseHeader || (exports.DevbaseHeader = DevbaseHeader = {}));
class BaseProcessor {
    context;
    constructor(context) {
        this.context = context;
    }
    buildHeaders(entityName, entityId) {
        const { appId, user } = this.context;
        const headers = {};
        if (entityName) {
            headers[DevbaseHeader.SECRET] = (0, utils_1.buildEntitySecret)({ appId, entityName, entityId });
        }
        if (user) {
            headers[DevbaseHeader.USER_ID] = user.id;
            headers[DevbaseHeader.USER_TOKEN] = user.token;
        }
        return headers;
    }
}
exports.BaseProcessor = BaseProcessor;
